### ISSUE TITLE
Inference produces no change: Gender transfer using CelebA-HQ and CycleGAN

### ISSUE THREAD

**Issue by arunasank** [junyanz/pytorch-CycleGAN-and-pix2pix#944]
Hi, I am using this to perform gender transfer on the CelebA-HQ dataset. Each image is of the size 1024 x 1024. I have classified the images into two folders trainA (male) and trainB (female). I am using the following command to train the model:

```
python3 train.py --dataroot ./datasets/male2femalehd/ --name male2femalehd_cyclegan --model cycle_gan --resize_or_crop scale_width_and_crop --fineSize 360 --loadSize 1024 --batch_size 15 --gpu_ids 0,1,2,3,4,5,6,7 --continue_train --norm instance --epoch <epoch>
```

I trained the model to 270 epochs, in 3 batches of 90 epochs each. 

1. When I run the model from a certain epoch using `--epoch <epoch>`, for example `--epoch 90`, it seems to run the epochs again starting from 1 and overwrite the checkpoints that already existed with new checkpoints, even though it's using the initial values from `90_net_D_A.pth`, `90_net_D_B.pth`, `90_net_G_A.pth`, `90_net_G_B.pth`. Is this expected?
2. After training the model for 270 epochs, I am running it on a couple of images from the training dataset by only using the `latest_net_G_A.pth` and renaming it to `latest_net_G.pth` using the following command 
```
python3 test.py --dataroot datasets/male2femalehd/testA/ --name male2femalehd_cyclegan --resize_or_crop scale_width --loadSize 1024 --model test --no_dropout
```

However, during inference, the input images show absolutely no change. Can you please help me in fine tuning my training? Are the epoch values I am passing problematic?

**Comment by arunasank** [comment]
## Fake image
![00010_fake_B](https://user-images.githubusercontent.com/3166852/75721456-678f6a00-5ca6-11ea-8980-31369d0d5552.png)

## Real image
![00010_real_A](https://user-images.githubusercontent.com/3166852/75721458-678f6a00-5ca6-11ea-8757-3d39115f5fc1.png)

**Comment by arunasank** [comment]
@junyanz sorry to ping you directly, but any advice would help. I am not sure where I am making a mistake in the training.

**Comment by arunasank** [comment]
Some related tickets for my reference as I research this:
* Information about loss curves: https://github.com/junyanz/pytorch-CycleGAN-and-pix2pix/issues/806
* Definition of generator and discriminator: https://github.com/junyanz/pytorch-CycleGAN-and-pix2pix/issues/941
* Learning rate and batch size: https://github.com/junyanz/pytorch-CycleGAN-and-pix2pix/issues/779
* Parameters common to train and test: https://github.com/junyanz/pytorch-CycleGAN-and-pix2pix/blob/master/options/base_options.py
* Architectures for generator and discriminator when the dataset contains images of large sizes: https://github.com/junyanz/pytorch-CycleGAN-and-pix2pix/issues/760
* D loss drops but G loss does not drop: https://github.com/junyanz/pytorch-CycleGAN-and-pix2pix/issues/858 and https://github.com/junyanz/pytorch-CycleGAN-and-pix2pix/search?q=D+loss+0&type=Issues

**Comment by arunasank** [comment]
Per some readings, and after some conversations with folks who know a little more about machine learning than I do, I am trying to increase the generator capacity and decrease the discriminator capacity, by using `--ngf 64 and --ndf 32` in my training code. I am restarting the training, so my training command is: `python3 train.py --dataroot ./datasets/male2femalehd/ --name male2femalehd_cyclegan --model cycle_gan --resize_or_crop scale_width_and_crop --fineSize 360 --loadSize 1024 --batch_size 10 --gpu_ids 0,1,2,3 --norm instance --save_epoch_freq 1 --netG resnet_9blocks --ndf 32`

**Comment by junyanz** [comment]
I recommend that you start your first experiment at a lower resolution model (`--load_size 286 --crop_size 256`) (We changed the flag name in the latest commit). The training will be faster and easier to debug. 

To continue the training from previous sessions, you need to use the flag `--continue_train`. If you can use `--epoch_count` to avoid overwriting your old models. See [here](https://github.com/junyanz/pytorch-CycleGAN-and-pix2pix/blob/master/docs/qa.md#can-i-continueresume-my-training-350-275-234-87) for more details.

Does the model make any changes to the training images during training? You may want to make sure that happens first.

**Comment by arunasank** [comment]
Thank you @junyanz. I'll incorporate the suggestions wrt to the training. Happy to report that with decreasing the discriminator capacity I am seeing vast improvements. However, even there, the first training I ran with the same settings (generator with 64 layers and discriminator with 32 layers) did not work, whereas the second training I am running is giving better results.

**Comment by arunasank** [comment]
To report further, downsampling the images to 256x256 and using full-sized images without cropped patches has made a significant difference to the training, and as @junyanz observed above, made it faster as well.

<img width="628" alt="PastedGraphic-1" src="https://user-images.githubusercontent.com/3166852/76876655-9aaa2f80-6848-11ea-99a7-f28c7e5d86b9.png">

<img width="629" alt="image" src="https://user-images.githubusercontent.com/3166852/76876690-aa297880-6848-11ea-9654-8a2969ff26e5.png">

**Comment by Rjaat** [comment]
Hi, @arunasank !!
Glad to see you got better results. Is it possible for you if you could provide me with the hyperparameter flags you used. Actually i am also facing the same issue as yours. Thank you

**Comment by arunasank** [comment]
@Rjaat Sorry this was a while ago, and I don't have these stored!
